### ISSUE TITLE
pragma parsing is broken by || (or operator)

### ISSUE THREAD

**Issue by moodysalem** [nomiclabs/hardhat#1089]
see https://github.com/abdk-consulting/abdk-libraries-solidity/issues/11

changing the pragma to >= and < works

the compiler config is:
```
  solidity: {
    version: '0.6.12',
    settings: {
      optimizer: {
        enabled: true,
        runs: 200,
      },
    },
  },
```

**Comment by fvictorio** [comment]
Thanks @moodysalem. This is a problem in the solidity parser, I opened an issue there: https://github.com/solidity-parser/parser/issues/35

**Comment by chanhosuh** [comment]
This issue should be closed, as it was resolved by the fix for #1130

**Comment by fvictorio** [comment]
Yes, this should be fixed. Please re-open if it isn't.
